### PR TITLE
Log changes to the ShardCoordinator ddata state

### PR DESCRIPTION
**Pull request by raboof**
The number of shards is configurable, in the order of magnitude of the number
of nodes in the cluster. Logging the ActorRef for each allocated shard is
useful to see on which node the shard is allocated.

### CODE REVIEW

**Review by chbatey — APPROVED** [review]

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala [hunk 1]
```diff
@@ -329,6 +329,8 @@ object ShardCoordinator {
         rememberEntities: Boolean = false)
         extends ClusterShardingSerializable {
 
+      override def toString = s"State($shards)"
```
Comment (helena) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: Or more to convention?
```suggestion
      override def toString = s"State[$shards]"
```

**Review by helena — APPROVED** [review]
Just one thought, but LGTM

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala (hunk 1 above)
Comment (raboof) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: This is 'case class style' ;) - indeed in the log messages I put `[]`around it as well :+1:

**Review by jrudolph — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
I'm somewhat worried that this will be huge log messages that are logged rather frequently. There can be 1000s of shards. Let's leave it as this for now but pay attention to complaints.

**Pull request merged**